lightningd/plugin_control.c

```diff
@@ -0,0 +1,135 @@
+#include <ccan/tal/path/path.h>
+#include <common/json_command.h>
+#include <common/jsonrpc_errors.h>
+#include <common/param.h>
+#include <dirent.h>
+#include <lightningd/plugin_control.h>
+#include <lightningd/plugin_hook.h>
+#include <unistd.h>
+
+/**
+ * A plugin command which permits to control plugins without restarting
+ * lightningd. It takes a command as first parameter (named 'control'
+ * to avoid ambuiguity), and eventually this control's parameter(s) as
+ * optional second and third parameters.
+ */
+static struct command_result *json_plugin_control(struct command *cmd,
+						const char *buffer,
+						const jsmntok_t *obj UNNEEDED,
+						const jsmntok_t *params)
+{
+	const char *control_command, *control_param, *control_second_param;
+	struct plugin *p;
+	struct json_stream *response;
+	bool plugin_found;
+
+	if (!param(cmd, buffer, params,
+		   p_req("command", param_string, &control_command),
+		   p_opt("parameter", param_string, &control_param),
+		   p_opt("second_parameter", param_string, &control_second_param),
+		   NULL))
+		return command_param_failed();
+
+	if (!control_param && !streq(control_command, "list")
+			&& !streq(control_command, "reload"))
+		return command_fail(cmd, JSONRPC2_INVALID_PARAMS,
+					"Missing a required parameter to the"
+					" plugin command \"%s\"", control_command);
+
+	if (streq(control_command, "stop")) {
+		plugin_found = false;
+		list_for_each(&cmd->ld->plugins->plugins, p, list) {
+			if (paths_match(p->cmd, control_param)) {
```

Comment:
Rather than export paths_match, we should export 'find_plugin' which does this loop for us?

Reply:
`find_plugin` would abort on find failure IIRC